convert2rhel/subscription.py

```diff
@@ -278,6 +287,93 @@ def args(self):
         return args
 
     def __call__(self):
+        """When called, this will register the system with subscription-manager."""
+        # Questions:
+        # * Should we check whether dbus is running?
+        # * If it's not running should we install and run it or should we fall back to
+        #   subscription-managr cli?
+        # * If we install it, do we need to revert that after the upgrade/rollback?
+        # * If we fallback to cli, do we still allow activation_key even though it is insecure?
+        # * Should checking for DBUS be done in either system_info or checks.py?
+        # * What should the return value of this code be?  Should cli non-zero retcodes raise an
+        #   exception instead of returning the code?
+        # TODO:
+        # * In _call_via_dbus(), we need to add more logging.  Call_via_cli lets
+        #   run_subprocess()/run_in_pty() do most of its logging but we don't have that in
+        #   _call_via_dbus()
+        if DBUS:
+            return self._call_via_dbus()
+        else:
+            return self._call_via_cli()
+
+    def _call_via_dbus(self):
+        """
+        Use dbus to register the system with subscription-manager.
+
+        Status of dbus on various platforms:
+            * RHEL6:
+                * DBUS is present but may not be installed.
+                * Install the dbus package and run /etc/rc.d/init.d/messagebus start
+                * dbus-python-0.83.0 is available
+            * RHEL7:
+                * dbus looks to be installed and run by default
+                * dbus-python-1.1.9 is available
+            * RHEL8:
+            * RHEL9:
+                * dbus looks to be installed and run by default
+                * python3-dbus-1.2.18 is available
+
+        .. seealso::
+            Documentation for the subscription-manager dbus API:
+            https://www.candlepinproject.org/docs/subscription-manager/dbus_objects.html#registerserver
+        """
+        system_bus = dbus.SystemBus()
+
+        if self.server_url:
+            connection_opts = {"host": self.server_url}
+        else:
+            # DBus seems to want empty string if there are no values even though the type is a dict
+            connection_opts = ""
+
+        # Create a new bus so we can talk to rhsm privately (For security:
+        # talking on the system bus might be eavesdropped in certain scenarios)
+        register_server = system_bus.get_object("com.redhat.RHSM1", "/com/redhat/RHSM1/RegisterServer")
+        address = register_server.Start("", dbus_interface="com.redhat.RHSM1.RegisterServer")
```

Comment:
the first parameter is the locale for the potential error messages; maybe it'd be better to pass the current locale of convert2rhel; the same applies to the other calls that take a locale as parameter

Reply:
Cool.  We always use the C locale for the things that we're calling so I will definitely add that to these function calls.

Reply:
Pushed a commit with all locale fields filled in.